### PR TITLE
UIORGS-48 remove falsy error notification (stripes-connect bug)

### PR DESCRIPTION
**Pull request by aliaksei-chumakou**
## Purpose
There is a bug in stripes connect when (a couple, actually), so there is a false error notification when you wait promises on DELETE API call.
https://issues.folio.org/browse/UIORGS-48

## Learning
Basically, 1 error - to not use `path` as function - there is an exception on resource refresh in stripes-connect.
2nd - to not wait `then` after `mutator.DELETE()`, since stripes connect calls GET on that resource right after delete and gets 404 response, so returns an `.catch` case

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by mkuklis** [comment]
@aliaksei-chumakou would you mind describing in couple steps how to reproduce this error? I will try to look into stripes-connect to address it. Thank you!

**Comment by aliaksei-chumakou** [comment]
@mkuklis Sure, 
1) As you can see in the code of this PR, previously it was `contactResource`, where path is function, not the string. And on call of mutator.put of another resource, it called refresh of all resource and failed trying to call startsWith on function in stripes-connect `refreshEpic` (string is expected). https://issues.folio.org/browse/STCON-48
2) after `deleteContact` in promise chaining it goes to catch since error comes because after delete api call there is resource refresh in stripes-connect - 404 comes from back-end. It looks like could be reproduced on any resource, just try to go into `mutator.DELETE().then()`
Basically, this PR just uses `finally` to get things work for demo purposes

**Comment by mkuklis** [comment]
Thank you @aliaksei-chumakou! I made couple changes in https://github.com/folio-org/stripes-connect/pull/88 which should hopefully address this issue.
